### DISCUSSION TITLE
Feat: Disable Markdown

### DISCUSSION THREAD

**Discussion by moritztim** [TheR1D/shell_gpt#412] [Ideas]
Please add the option to disable markdown through a flag. I'm piping the output directly and I need plaintext.
*I do love that you implemented markdown tho :D*

**Comment by moritztim** [comment]
It's a small modification here, if someone wants to beat me to it
https://github.com/TheR1D/shell_gpt/blob/482ec9dbaa8a12b316f93d6f6efb8c476d704d46/sgpt/handlers/handler.py#L62C1-L65C62

**Reply by edshamis** [reply]
It would be very useful addition

**Comment by gitmpr** [comment]
this can be achieved most easily through creating a custom role

```
$ sgpt --create-role nomarkdown
Enter role description: Provide only code in plain text format without Markdown formatting or any leading whitespace
$ sgpt --role "nomarkdown" "how would you create a bash script, make it executable, and then execute it?"
```

**Reply by moritztim** [reply]
I don't have it set up right now, could you test if that provides a clean output, without additional whitespace? That was my main problem.

**Reply by gitmpr** [reply]
yep this works well
```
$ sgpt --role "nomarkdown" "can you provide me all the steps to create a bash script, then make it executable, and then how to execute it?"
1. Create a new bash script file using a text editor, for example, `nano`:
nano script.sh

2. Write your bash script inside the file. For example:
#!/bin/bash
echo "Hello, World!"

3. Save the file and exit the text editor (in `nano`, press `CTRL+X`, then `Y`, then `Enter`).

4. Make the script executable by changing its permissions:
chmod +x script.sh

5. Execute the script:
./script.sh

$ sgpt --show-role nomarkdown
You are nomarkdown
Provide code in plain text format without Markdown formatting or any leading whitespace
```

edit: another thing you can try is uing the `--no-md` flag

**Reply by moritztim** [reply]
Thanks for testing!

> another thing you can try is uing the --no-md flag

Oh, well that solves it.
